example/kotlinlib/module/6-annotation-processors/build.mill

```diff
@@ -0,0 +1,44 @@
+package build
+
+import mill._, kotlinlib._
+import java.io.File
+
+// The Kotlin compiler requires plugins to be passed explicitly. To do this, you can define
+// a module to contain the exact annotation processors you want, and pass
+// in `-Xplugin` to `kotlincOptions`:
+
+object foo extends KotlinModule {
+
+  def kotlinVersion = "1.9.24"
+
+  def ivyDeps = super.ivyDeps() ++ Agg(
+    ivy"org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.3"
+  )
+
+  def processors = T {
+    defaultResolver().resolveDeps(
+      Agg(
+        ivy"org.jetbrains.kotlin:kotlin-serialization-compiler-plugin:1.9.24"
+      )
+    )
+  }
+
+  def kotlincOptions = super.kotlincOptions() ++ Seq(
+    s"-Xplugin=${processors().map(_.path).head}"
+  )
+
+  object test extends KotlinModuleTests with TestModule.Junit5 {
+    def ivyDeps = super.ivyDeps() ++ Agg(
+      ivy"io.kotest:kotest-runner-junit5-jvm:5.9.1"
+    )
+  }
+}
+
+/**
+ * Usage
+ *
```

Comment:
The `Usage` comment has to follow the format of the others, this one was probably auto-reformatted which would cause the example tester to fail